python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -0,0 +1,79 @@
+# See COPYRIGHT file at the top of the source tree.
+
+from __future__ import absolute_import, division, print_function
+
+import abc
+import numpy as np
+
+from lsst.afw import table
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+
+__all__ = ["BaseSourceSelectorConfig", "BaseSourceSelectorTask", "sourceSelectorRegistry"]
+
+
+class BaseSourceSelectorConfig(pexConfig.Config):
+    badFlags = pexConfig.ListField(
+        doc = "List of flags which cause a source to be rejected as bad",
+        dtype = str,
+        default = [
+            "base_PixelFlags_flag_edge",
+            "base_PixelFlags_flag_interpolatedCenter",
+            "base_PixelFlags_flag_saturatedCenter",
+            "base_PixelFlags_flag_crCenter",
+            "base_PixelFlags_flag_bad",
+            "base_PixelFlags_flag_interpolated",
+        ],
+    )
+
+
+class BaseSourceSelectorTask(pipeBase.Task):
+    """!Base class for source selectors
+
+    Register all source selectors with the sourceSelectorRegistry using:
+        sourceSelectorRegistry.register(name, class)
+    """
+    __metaclass__ = abc.ABCMeta
+
+    ConfigClass = BaseSourceSelectorConfig
+    _DefaultName = "sourceSelector"
+
+    def __init__(self, **kwargs):
+        """!Initialize a source selector."""
+        pipeBase.Task.__init__(self, **kwargs)
+
+    def run(self, sourceCat, maskedImage=None, **kwargs):
+        """!Select sources and return them.
+
+        @param[in] sourceCat  catalog of sources that may be sources (an lsst.afw.table.SourceCatalog)
+        @param[in] maskedImage  the maskedImage containing the sources, for plotting.
+
+        @return an lsst.pipe.base.Struct containing:
+        - sourceCat  catalog of sources that were selected
+        """
+        return self.selectSources(maskedImage=maskedImage, sourceCat=sourceCat, **kwargs)
+
+    @abc.abstractmethod
+    def selectSources(self, sourceCat, matches=None):
+        """!Return a catalog of sources: a subset of sourceCat.
+
+        @param[in] sourceCat  catalog of sources that may be sources (an lsst.afw.table.SourceCatalog)
+
+        @return a pipeBase.Struct containing:
+        - sourceCat  a catalog of sources
+        """
+
+        # NOTE: example implementation, returning all sources that have no bad flags set.
+        result = table.SourceCatalog(sourceCat.table)
+        for source in sourceCat:
+            if not self._isBad(source):
+                result.append(source)
+        return pipeBase.Struct(sourceCat=result)
+
+    def _isBad(self, source):
+        """Return True if any of config.badFlags are set for this source."""
+        return reduce(lambda x, y: np.logical_or(x, source.get(y)), self.config.badFlags, False)
```

Comment:
It would take a few more lines, but why not write:
for field in self.config.badFlags:
    if source.get(field) is True:
        return True
else:
    return False